weave-js/src/components/Button/Button.tsx

```diff
@@ -4,153 +4,155 @@
  * https://www.figma.com/file/01KWBdMZg5QM9SRS1pQq0z/Design-System----Robot-Styles?type=design&node-id=5956-31813&mode=design&t=Gm4WWGWwgjdfUUTe-0
  */
 
-import {TooltipContentProps} from '@radix-ui/react-tooltip';
+import {type TooltipContentProps} from '@radix-ui/react-tooltip';
 import classNames from 'classnames';
-import React, {ReactElement, useState} from 'react';
+import React, {
+  CSSProperties,
+  forwardRef,
+  type ReactElement,
+  useState,
+} from 'react';
 import {twMerge} from 'tailwind-merge';
 
-import {Icon, IconName} from '../Icon';
+import {Icon, type IconName} from '../Icon';
 import * as Tooltip from '../RadixTooltip';
 import {Tailwind} from '../Tailwind';
-import {ButtonSize, ButtonVariant} from './types';
+import type {ButtonSize, ButtonVariant} from './types';
 
-export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
-  /**
-   * Shorthand alias for `startIcon` since most button icons are startIcons.
-   * If both props are provided, `startIcon` value will be used out of the two.
-   */
+type IconButtonProps = {
   icon?: IconName;
-  /**
-   * Icon displayed before the children
-   */
+  children?: never;
+  startIcon?: never;
+  endIcon?: never;
+  tooltip: string;
```

Comment:
the only thing is, if the aria label and the tooltip _are_ in fact the same, then we should probably use [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) to label the button by the tooltip contents

Reply:
Interesting point! I guess it depends on _how_ the tooltip is rendered -- if it is detached from the button then the aria-labelledby make sense on the button, but we could also just assign the tooltip's value to aria-label on the button.

As for design not wanting a tooltip in certain areas, I'd be curious when this is applicable. If someone uses a screen reader b/c of poor vision or something, an aria label is needed to understand what an icon button does in the same way that it would read the children of the button. The flip side of someone without those impairments is that they need the tooltip to understand what the icon button does, and the children of the button could just be read. 🤔 I'd defer to WCAG specs & design team on this topic but we can definitely remove the necessity for it right now.